client.go

```diff
@@ -423,3 +427,62 @@ func (c *Client) String() string {
 func (c *Client) setHeader(key, val string) {
 	c.headers[key] = val
 }
+
+func (c *Client) getEventSource(fql *Query, opts ...QueryOptFn) (*EventSource, error) {
+	res, err := c.Query(fql, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	token, ok := res.Data.(EventSource)
+	if !ok {
+		return nil, fmt.Errorf("query should return a fauna.EventSource but got %T", res.Data)
+	}
+
+	return &token, nil
+}
+
+// FeedFromQuery opens an event feed from the event source returned by the [fauna.Query].
+func (c *Client) FeedFromQuery(fql *Query, opts ...QueryOptFn) (*EventFeed, error) {
+	token, err := c.getEventSource(fql, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return newEventFeed(c, *token)
+}
+
+// FeedFromQueryWithStartTime initiates an event from the event source returned by the [fauna.Query] with custom options
+func (c *Client) FeedFromQueryWithStartTime(fql *Query, start FeedStartFn, opts ...QueryOptFn) (*EventFeed, error) {
+	token, err := c.getEventSource(fql, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return newEventFeed(c, *token, start)
+}
+
+// FeedFromQueryWithCursor initiates an event from the event source returned by the [fauna.Query] with custom options
+func (c *Client) FeedFromQueryWithCursor(fql *Query, cursor FeedStartFn, opts ...QueryOptFn) (*EventFeed, error) {
+	token, err := c.getEventSource(fql, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return newEventFeed(c, *token, cursor)
+}
+
+// Feed opens an event feed from the event source
+func (c *Client) Feed(stream EventSource) (*EventFeed, error) {
+	return newEventFeed(c, stream)
+}
+
+// FeedWithStartTime opens an event feed from the event source with options
+func (c *Client) FeedWithStartTime(stream EventSource, start FeedStartFn) (*EventFeed, error) {
```

Comment:
```suggestion
func (c *Client) FeedWithStartTime(stream EventSource, start EventFeedStartTime) (*EventFeed, error) {
```

or

```suggestion
func (c *Client) FeedWithStartTime(stream EventSource, startTs int64) (*EventFeed, error) {
```

Reply:
I opt'd to use `time.Time` and handle the `UnixMicro()` within, would you rather me stick to int64?